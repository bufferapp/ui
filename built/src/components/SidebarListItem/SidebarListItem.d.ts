import React from 'react';
import PropTypes from 'prop-types';
declare const SidebarListItem: {
    ({ title, icon, onItemClick, badges, badgeIcon, selected, user, className, }: {
        title: any;
        icon: any;
        onItemClick: any;
        badges: any;
        badgeIcon: any;
        selected: any;
        user: any;
        className: any;
    }): React.JSX.Element;
    propTypes: {
        id: PropTypes.Requireable<string>;
        className: PropTypes.Requireable<string>;
        title: PropTypes.Validator<string>;
        icon: PropTypes.Requireable<PropTypes.ReactNodeLike>;
        onItemClick: PropTypes.Validator<(...args: any[]) => any>;
        badges: PropTypes.Requireable<string | number | boolean | PropTypes.ReactElementLike | PropTypes.ReactNodeArray>;
        badgeIcon: PropTypes.Requireable<PropTypes.ReactNodeLike>;
        selected: PropTypes.Requireable<boolean>;
        user: PropTypes.Requireable<PropTypes.InferProps<{
            id: PropTypes.Requireable<string>;
            name: PropTypes.Requireable<string>;
            handle: PropTypes.Requireable<string>;
            profileImageUrl: PropTypes.Requireable<string>;
            fallbackUrl: PropTypes.Requireable<string>;
            network: PropTypes.Requireable<string>;
        }>>;
    };
    defaultProps: {
        id: string;
        icon: null;
        badgeIcon: null;
        badges: null;
        selected: null;
        user: null;
        className: null;
    };
};
export default SidebarListItem;
//# sourceMappingURL=SidebarListItem.d.ts.map