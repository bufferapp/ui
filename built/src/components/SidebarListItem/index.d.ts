export { default } from './SidebarListItem';
//# sourceMappingURL=index.d.ts.map