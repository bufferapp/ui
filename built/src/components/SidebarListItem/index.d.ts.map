{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../src/components/SidebarListItem/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,mBAAmB,CAAA"}