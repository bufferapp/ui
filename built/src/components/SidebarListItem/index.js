export { default } from './SidebarListItem';
//# sourceMappingURL=index.js.map