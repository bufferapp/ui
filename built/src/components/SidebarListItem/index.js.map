{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../src/components/SidebarListItem/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,mBAAmB,CAAA","sourcesContent":["export { default } from './SidebarListItem'\n"]}