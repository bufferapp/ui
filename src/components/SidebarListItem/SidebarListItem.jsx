import React from 'react';

import PropTypes from 'prop-types';
import {
  Badge,
  BadgeIconContainer,
  Handle,
  IconContainer,
  ItemStyled,
  LabelContainer,
  LabelStyled,
  NameHandleWrapper,
} from './style';
import Avatar from '../Avatar/Avatar';

const SidebarListItem = ({
  title,
  icon,
  onItemClick,
  badges,
  badgeIcon,
  selected,
  user,
  className,
}) => (
  <ItemStyled
    onClick={() => onItemClick()}
    hasUser={user}
    selected={selected}
    className={className}
  >
    {icon && <IconContainer selected={selected}>{icon}</IconContainer>}
    <LabelContainer>
      {user ? (
        <React.Fragment>
          <Avatar
            src={user.profileImageUrl}
            fallbackUrl={user.fallbackUrl}
            alt={user.name}
            size="small"
            type="social"
            network={user.network}
          />
          <NameHandleWrapper>
            <LabelStyled
              type="label"
              title={title}
              hasUser={user}
              selected={selected}
            >
              {user.name}
            </LabelStyled>
            <Handle selected={selected}>{user.handle}</Handle>
          </NameHandleWrapper>
        </React.Fragment>
      ) : (
        <LabelStyled type="label" title={title} selected={selected}>
          {title}
        </LabelStyled>
      )}
    </LabelContainer>

    {!badgeIcon && badges && <Badge selected={selected}>{badges}</Badge>}
    {badgeIcon && (
      <BadgeIconContainer selected={selected}>{badgeIcon}</BadgeIconContainer>
    )}
  </ItemStyled>
);

SidebarListItem.propTypes = {
  /** The id of the element */
  id: PropTypes.string,
  /** The className of the element */
  className: PropTypes.string,
  /** What the label will say */
  title: PropTypes.string.isRequired,
  /** An icon either from this library or a node of your choice */
  icon: PropTypes.node,
  /** A function to perform when the item is clicked */
  onItemClick: PropTypes.func.isRequired,
  /** A string or Node to display at the far right side of the item */
  badges: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  /** An icon either from this library or a node of your choice */
  badgeIcon: PropTypes.node,
  /** Whether the item is currently selected */
  selected: PropTypes.bool,
  /** A user object if you'd like the item to display the user Avatar, social network and handle */
  user: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    handle: PropTypes.string,
    profileImageUrl: PropTypes.string,
    fallbackUrl: PropTypes.string,
    network: PropTypes.oneOf([
      'facebook',
      'twitter',
      'instagram',
      'linkedin',
      'google',
      'pinterest',
      'tiktok',
      'googlebusiness',
      'startPage',
      'mastodon',
      'youtube',
    ]),
  }),
};

SidebarListItem.defaultProps = {
  id: '',
  icon: null,
  badgeIcon: null,
  badges: null,
  selected: null,
  user: null,
  className: null,
};

export default SidebarListItem;
