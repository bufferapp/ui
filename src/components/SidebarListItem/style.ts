import styled from 'styled-components'
import {
  blueLightest50,
  grayDarker,
  grayLight50,
  gray,
  grayDark,
  blue,
} from '../style/colors'
import Text from '../Text/Text'

export const Badge = styled.span`
  margin-left: auto;
  font-family: Roboto;
  font-style: normal;
  font-weight: bold;
  font-size: 12px;
  line-height: 14px;
  text-align: right;
  margin-right: 8px;
  color: ${(props): string => (props.selected ? blue : grayDarker)};
`

export const NameHandleWrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 16px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
`

export const Handle = styled.span`
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 14px;
  letter-spacing: -0.2px;
  color: ${grayDark};
`

export const LabelContainer = styled.div`
  display: flex;
  align-items: center;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  margin-right: 8px;
`

export const LabelStyled = styled(Text)`
  font-size: ${(props): string => (props.hasUser ? '13px' : '15px')};
  font-weight: ${(props): string => (props.hasUser ? '500' : '700')};
  line-height: ${(props): string => (props.hasUser ? '20px' : '24px')};
  color: ${(props): string => (props.selected ? blue : grayDarker)};
`

export const IconContainer = styled.span`
  margin-right: 8px;
  display: inline-block;
  height: 16px;
  width: 16px;
  display: flex;
  justify-content: center;
  svg {
    fill: ${(props): string => (props.selected ? blue : gray)};
  }
`

export const ItemStyled = styled.li`
  list-style-type: none;
  display: flex;
  align-items: center;
  cursor: pointer;
  border-radius: 4px;
  padding: 8px 0 8px 8px;
  background-color: ${(props) =>
    props.selected ? blueLightest50 : 'transparent'};

  &:hover {
    background-color: ${(props) =>
      props.selected ? blueLightest50 : grayLight50};

    ${Badge}, ${LabelStyled} {
      color: ${(props): string => (props.selected ? blue : grayDarker)};
    }

    ${IconContainer} svg {
      fill: ${(props): string => (props.selected ? blue : gray)};
    }
  }
`

export const BadgeIconContainer = styled(IconContainer)`
  margin-left: auto;
  svg {
    fill: ${(props): string => (props.selected ? gray : 'currentColor')};
  }
`
