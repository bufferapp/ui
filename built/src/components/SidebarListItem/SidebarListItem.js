import React from 'react';
import PropTypes from 'prop-types';
import { Badge, BadgeIconContainer, Handle, IconContainer, ItemStyled, LabelContainer, LabelStyled, NameHandleWrapper, } from './style';
import Avatar from '../Avatar/Avatar';
const SidebarListItem = ({ title, icon, onItemClick, badges, badgeIcon, selected, user, className, }) => (React.createElement(ItemStyled, { onClick: () => onItemClick(), hasUser: user, selected: selected, className: className },
    icon && React.createElement(IconContainer, { selected: selected }, icon),
    React.createElement(LabelContainer, null, user ? (React.createElement(React.Fragment, null,
        React.createElement(Avatar, { src: user.profileImageUrl, fallbackUrl: user.fallbackUrl, alt: user.name, size: "small", type: "social", network: user.network }),
        React.createElement(NameHandleWrapper, null,
            React.createElement(LabelStyled, { type: "label", title: title, hasUser: user, selected: selected }, user.name),
            React.createElement(Handle, { selected: selected }, user.handle)))) : (React.createElement(LabelStyled, { type: "label", title: title, selected: selected }, title))),
    !badgeIcon && badges && React.createElement(Badge, { selected: selected }, badges),
    badgeIcon && (React.createElement(BadgeIconContainer, { selected: selected }, badgeIcon))));
SidebarListItem.propTypes = {
    id: PropTypes.string,
    className: PropTypes.string,
    title: PropTypes.string.isRequired,
    icon: PropTypes.node,
    onItemClick: PropTypes.func.isRequired,
    badges: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
    badgeIcon: PropTypes.node,
    selected: PropTypes.bool,
    user: PropTypes.shape({
        id: PropTypes.string,
        name: PropTypes.string,
        handle: PropTypes.string,
        profileImageUrl: PropTypes.string,
        fallbackUrl: PropTypes.string,
        network: PropTypes.oneOf([
            'facebook',
            'twitter',
            'instagram',
            'linkedin',
            'google',
            'pinterest',
            'tiktok',
            'googlebusiness',
            'startPage',
            'mastodon',
            'youtube',
        ]),
    }),
};
SidebarListItem.defaultProps = {
    id: '',
    icon: null,
    badgeIcon: null,
    badges: null,
    selected: null,
    user: null,
    className: null,
};
export default SidebarListItem;
//# sourceMappingURL=SidebarListItem.js.map