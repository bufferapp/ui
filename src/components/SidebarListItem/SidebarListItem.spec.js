/* eslint-disable react/jsx-filename-extension */
import snap from 'jest-auto-snapshots'
import 'jest-styled-components'

import SidebarListItem from './SidebarListItem'

snap(SidebarListItem, './SidebarListItem.jsx')
