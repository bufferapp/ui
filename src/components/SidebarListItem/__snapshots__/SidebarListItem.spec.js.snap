// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > SidebarListItem Matches snapshot when boolean prop "selected" is set to: "false" 1`] = `
<ForwardRef(styled.li)
  className="jest-auto-snapshots String Fixture"
  hasUser={
    Object {
      "fallbackUrl": "jest-auto-snapshots String Fixture",
      "handle": "jest-auto-snapshots String Fixture",
      "id": "jest-auto-snapshots String Fixture",
      "name": "jest-auto-snapshots String Fixture",
      "network": "facebook",
      "profileImageUrl": "jest-auto-snapshots String Fixture",
    }
  }
  onClick={[Function]}
  selected={false}
>
  <ForwardRef(styled.span)
    selected={false}
  >
    <NodeFixture />
  </ForwardRef(styled.span)>
  <ForwardRef(styled.div)>
    <React.Fragment>
      <Avatar
        alt="jest-auto-snapshots String Fixture"
        fallbackUrl="jest-auto-snapshots String Fixture"
        network="facebook"
        size="small"
        src="jest-auto-snapshots String Fixture"
        type="social"
      />
      <ForwardRef(styled.div)>
        <Text
          color="grayDarker"
          hasError={false}
          title="jest-auto-snapshots String Fixture"
          type="label"
        >
          jest-auto-snapshots String Fixture
        </Text>
        <ForwardRef(styled.span)
          selected={false}
        >
          jest-auto-snapshots String Fixture
        </ForwardRef(styled.span)>
      </ForwardRef(styled.div)>
    </React.Fragment>
  </ForwardRef(styled.div)>
  <ForwardRef(Styled(styled.span))
    selected={false}
  >
    <NodeFixture />
  </ForwardRef(Styled(styled.span))>
</ForwardRef(styled.li)>
`;

exports[`jest-auto-snapshots > SidebarListItem Matches snapshot when passed all props 1`] = `
<ForwardRef(styled.li)
  className="jest-auto-snapshots String Fixture"
  hasUser={
    Object {
      "fallbackUrl": "jest-auto-snapshots String Fixture",
      "handle": "jest-auto-snapshots String Fixture",
      "id": "jest-auto-snapshots String Fixture",
      "name": "jest-auto-snapshots String Fixture",
      "network": "facebook",
      "profileImageUrl": "jest-auto-snapshots String Fixture",
    }
  }
  onClick={[Function]}
  selected={true}
>
  <ForwardRef(styled.span)
    selected={true}
  >
    <NodeFixture />
  </ForwardRef(styled.span)>
  <ForwardRef(styled.div)>
    <React.Fragment>
      <Avatar
        alt="jest-auto-snapshots String Fixture"
        fallbackUrl="jest-auto-snapshots String Fixture"
        network="facebook"
        size="small"
        src="jest-auto-snapshots String Fixture"
        type="social"
      />
      <ForwardRef(styled.div)>
        <Text
          color="white"
          hasError={false}
          title="jest-auto-snapshots String Fixture"
          type="label"
        >
          jest-auto-snapshots String Fixture
        </Text>
        <ForwardRef(styled.span)
          selected={true}
        >
          jest-auto-snapshots String Fixture
        </ForwardRef(styled.span)>
      </ForwardRef(styled.div)>
    </React.Fragment>
  </ForwardRef(styled.div)>
  <ForwardRef(Styled(styled.span))
    selected={true}
  >
    <NodeFixture />
  </ForwardRef(Styled(styled.span))>
</ForwardRef(styled.li)>
`;

exports[`jest-auto-snapshots > SidebarListItem Matches snapshot when passed only required props 1`] = `
<ForwardRef(styled.li)
  className={null}
  hasUser={null}
  onClick={[Function]}
  selected={null}
>
  <ForwardRef(styled.div)>
    <Text
      color="grayDarker"
      hasError={false}
      title="jest-auto-snapshots String Fixture"
      type="label"
    >
      jest-auto-snapshots String Fixture
    </Text>
  </ForwardRef(styled.div)>
</ForwardRef(styled.li)>
`;
